flink-connector-kafka/src/test/java/org/apache/flink/connector/kafka/source/KafkaSourceITCase.java

```diff
@@ -369,6 +370,38 @@ public void testConsumingTopicWithEmptyPartitions() throws Throwable {
                             WatermarkStrategy.noWatermarks(),
                             "testConsumingTopicWithEmptyPartitions"));
         }
+
+        @Test
+        @Timeout(value = 10)
```

Comment:
Why are you using Timeout here? We usually discourage the usage as any infra hiccup will result in test failure. Instead we rely on a global timeout where small infra hiccups are absorbed.

Reply:
I used the Timeout to prevent infinite blocking and fail the test if it enters an infinite loop. 
Since I'm not familiar with the project's global timeout — could you elaborate more about it? 
Where is it configured? 
If global timeout is set then it should be enough but when I tested this method on the current main branch without the Timeout, it didn't fail but just hung.

Reply:
The global timeout is applied during CI only. So after X minutes, the CI build will fail. If you test it locally, you have to abort manually (or you use the scripts to run the build as CI does).

Reply:
Got it, thanks for the explanation. I've deleted all custom timeout logic in test codes